### PR TITLE
Tasks

### PR DESCRIPTION
**Pull request by danisidiatullin**

### CODE REVIEW

File: src/models.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class ChoiceType(types.TypeDecorator):
+    impl = types.String
+
+    def __init__(self, choices, **kw):
+        self.choices = dict(choices)
+        super(ChoiceType, self).__init__(**kw)
+
+    def process_bind_param(self, value, dialect):
```
Comment (siarheipuhach) [inline src/models.py:27]: `dialect` does not look like used here.

File: src/schemas.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    id: int
+
+    class Config:
+        orm_mode = True
+
+
+class StatusEnum(str, Enum):
+    started = "started"
+    pending = "pending"
+    finished = "finished"
```
Comment (siarheipuhach) [inline src/schemas.py]: usually capital letters used like `STARTED = 'started'` so `StatusEnum.STARTED`

File: src/services.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    db.refresh(user_obj)
+
+    return user_obj
+
+
+async def authenticate_user(email: str, password: str, db: orm.Session):
+    user = await get_user_by_email(email, db)
+
+    if not user:
+        return False
```
Comment (siarheipuhach) [inline src/services.py]: I would return `None` instead of `False`.

File: src/models.py (hunk 1 above)
Comment (danisidiatullin) [reply src/models.py:27]: Yes, I agree. But the method `process_bind_param` is from `sqlalchemy. types` with these params and `NotImplementedError` so IDE highlights(warning) it without the `dialect` param.

**Pull request merged**